dpnp/backend_mathematical.pyx

```diff
@@ -221,7 +221,91 @@ cpdef dpnp_sum(dparray x1):
     """ Numpy interface inconsistency """
     return_type = numpy.dtype(numpy.int64) if (x1.dtype == numpy.int32) else x1.dtype
 
-    return return_type.type(result[0])
+    cdef dparray result_ = dparray((1,), dtype=return_type)
+    result_[0] = return_type.type(result)
+
+    return result_
+
+
+cpdef dparray dpnp_sum(dparray input, axis=None):
+    if axis is None:
+        return _dpnp_sum(input)
+
+    cdef long size_input = input.size
+    cdef dparray_shape_type shape_input = input.shape
+
+    if input.dtype == numpy.int32:
+        res_type = numpy.int64
+    else:
+        res_type = input.dtype
+
+    if isinstance(axis, int):
+        axis_ = tuple([axis])
+    else:
+        axis_ = axis
+
+    output_shape = dparray(len(shape_input) - len(axis_), dtype=numpy.int64)
+    ind = 0
+    for id, shape_axis in enumerate(shape_input):
+        if id not in axis_:
+            output_shape[ind] = shape_axis
+            ind += 1
+
+    cdef long prod = 1
+    for i in range(len(output_shape)):
+        if output_shape[i] != 0:
```

Comment:
Why this check is needed? If we have `0` in dimension - no memory created for this input array.
Also, Please create (or use if it exists) a function to do such shape calculation somewhere in https://github.com/IntelPython/dpnp/blob/e3d18674e64ba2c65c47fea0e3c4d0346898d532/dpnp/dpnp_utils.pyx#L157